### DISCUSSION TITLE
OPCUA Array's Visualization in Dashboard

### DISCUSSION THREAD

**Discussion by nicoleu23** [apache/streampipes#2736] [Q&A]
Hi all,
I am using the OPCUA (Read) Adapter, I am able to view in the dashboard different types of variables except for variables of the type array. It does not seem to provide a live preview for arrays, same thing for the data explorer.

Is it that streampipes or OPCUA is only restricted to some types of variables, or is it something that I am missing on my side?

Thank you, it would really be of great pleasure to use streampipes for the project I am working on.

Here are some screenshots.

Thank you,

Nicole
![Capture d'écran 2024-04-16 112620](https://github.com/apache/streampipes/assets/158181666/40798d4a-e2dc-49db-95c1-c11ca77797ee)
![Capture d'écran 2024-04-16 113014](https://github.com/apache/streampipes/assets/158181666/39e55f87-6822-4e7c-8d6d-aa528f3b4428)
![Capture d'écran 2024-04-16 113235](https://github.com/apache/streampipes/assets/158181666/67bade51-7e60-4432-8579-28d4af351417)
![Capture d'écran 2024-04-16 114535](https://github.com/apache/streampipes/assets/158181666/7aa25c51-862a-4261-bccf-54aec6878aac)

**Comment by bossenti** [comment]
Hi @nicoleu23,

we already had a discussion with respect to arrays in OPC UA data: https://github.com/apache/streampipes/discussions/1650#discussioncomment-6170814
I'd assume this is about the same problem as yours?
There was already an issue for this https://github.com/apache/streampipes/issues/1693, but this might have been mistakenly closed.

**Comment by nicoleu23** [comment]
Hi @bossenti ,

Thank you for your reply and for directing me to the issues. I just went through the respective issues and it my exact problem as well.
Do you by any chance know if there have been any possible update about this issue ?

Thank you,
Nicole

**Reply by bossenti** [reply]
Okay, thanks for checking!
I will reopen the issue so we can work on a fix.
Unfortunately, I can't give any estimate when the fix will be available.

**Reply by nicoleu23** [reply]
Thank you for the possibility to reopen the issue. I will be looking forward to the fix.
